src/components/button/code/templates/buttonTemplate.jsx

```diff
@@ -2,19 +2,16 @@ var React = require('react');
 var classNames = require('classnames');
 
 module.exports = function(props) {
-  var classes = {
-    'btn': true,
-    'btn-primary': props.primary,
-    'btn-secondary': props.secondary,
-    'btn-success': props.success,
-    'btn-info': props.info,
-    'btn-warning': props.warning,
-    'btn-danger': props.danger,
-    'btn-block': props.block,
-    'small': props.small,
-    'large': props.large,
-    'extra-large': props.extraLarge
-  };
+  var classes = ['component-button'];
+
+  if(props.size) {
+    classes.push(props.size);
+  }
+
+  if(props.purpose) {
+    classes.push(props.purpose);
+  }
+
   return (
     <button className={classNames(classes)} disabled={props.disabled} type={props.type}>
       {props.children}
```

Comment:
There should _probably_ be a test that renders the component into the document with `TestUtils` and then uses `$` to query the DOM and make sure it smells like what the CSS expects (or other JS expects - e.g. for binding events).